python/location/lbs/activity/audioset/dataprocessing/audio_processing.py

```diff
@@ -0,0 +1,262 @@
+# The downloaded audios would be clipped into 10 seconds periods
+# The downloaded audio segments are then processed using Librosa to extract features
+# Stored in eval_segments, balanced_train, and unbalanced_train separately.
+
+# Usage: python3 audio_prep.py <label>
+
+
+from __future__ import unicode_literals
+import youtube_dl
+import csv
+import json
+from pydub import AudioSegment
+import os
+import shutil
+from os import listdir
+from os.path import isfile, join
+import sys
+import numpy as np
+import librosa
+import argparse
+import numpy as np
+
+file_structure = "python/location/lbs/activity/audioset/dataprocessing/"
+referDoc = "ReferDoc/"
+downloaded_audio = []
+feature_dict = {}
+
+
+def setup_commandline_args():
+  parser = argparse.ArgumentParser(description='Process dataset.csv to input into TensorFlow.')
+  parser.add_argument('-d', '--datasets', required=True, type=str, nargs='+')
+  parser.add_argument('-r', '--redo', action='store_true', required=False)
+  parser.add_argument('-l', '--labels', required=True, type=str, nargs='+')
+  args = parser.parse_args()
+  print(args)
+  return args
+
+
+def parse_csv(csv_file):
+  # Extract all labeled audio info from csv
+  # input: str csv_file
+  # output: list audio_list
+  audio_list = []
+  with open(csv_file, newline='') as csvfile:
+    info = csv.reader(csvfile, delimiter=' ', quotechar='|')
+    for row in info:
+      r = ','.join(row)
+      if '#' not in r:
+        r_lst = r.split(',')
+        url, start_time, end_time = r_lst[0], float(r_lst[2]), float(r_lst[4])
+        audio_list.append([url, start_time, end_time])
+  return audio_list
+
+
+def label_id_search(label):
+  # Reads in the label, and finds the corresponding id in ontology.json
+  # input: str label
+  # output: str label_encode
+  with open(referDoc + "ontology.json") as f:
+    labelInfo = json.load(f)
+  for info in labelInfo:
+    if info["name"] == label:
+      return info["id"]
+  return None
+
+
+def download_from_list(dest_dir, audio_list, redo):
+  # Download labeled audio segments from given audio_list to a dest folder
+  # input: str dest_dir, list audio_list, str video_id
+  # output: None
+  for video_id, start_time, end_time in audio_list:
+    success = download(dest_dir, video_id, redo)
+    if success:
+      chop_audio(dest_dir, video_id, start_time, end_time)
+
+
+def download(dest_dir, video_id, redo):
+  # Download labeled audio from given url to a dest folder
+  # input: str dest_dir, str video_id
+  # output: boolean
+  try:
+    os.mkdir(dest_dir)
+  except OSError as error:
+    print(error)
+
+  # check to see if the video_id has already been downloaded and skip the download if it is already
+  # downloaded unless the -r, --redo flag has been passed
+  already_downloaded = isfile(dest_dir + '/sliced_' + video_id + '.wav')
+  print(already_downloaded and not redo)
+  if already_downloaded: # implement redo option
+    print('Already processed video')
```

Comment:
consider logging instead of printing:
https://abseil.io/docs/python/guides/logging